lib/drivers/tandemTslimDriver.js

```diff
@@ -1133,50 +1131,89 @@ module.exports = function (config) {
   };
 
   var buildTimeChangeRecords = function (data, postrecords) {
-    var timeChangeLogs = filterLogEntries([PUMP_LOG_RECORDS.LID_TIME_CHANGED, PUMP_LOG_RECORDS.LID_DATE_CHANGED],
-      data.log_records);
-
-    timeChangeLogs.forEach(function (change) {
-      var change_ts = change.header_ts;
-      //console.log(change);
-      var fromTime = change_ts - sundial.floor(change_ts, 'day', cfg.timezone);
-      var toTime = fromTime;
-      var fromDay = sundial.floor(change_ts, 'day', cfg.timezone); //TODO this doesn't appear to actually work because of UTC issues.
-      var toDay = fromDay;
-      if (change.header_id == PUMP_LOG_RECORDS.LID_DATE_CHANGED.value) {
-        toDay = BASE_TIME + change.date_after * sundial.MIN_TO_MSEC * 60 * 24;
-        fromDay = BASE_TIME + change.date_prior * sundial.MIN_TO_MSEC * 60 * 24;
+    var timeChangeLogs = filterLogEntries(
+      [PUMP_LOG_RECORDS.LID_TIME_CHANGED],
+      data.log_records
+    );
+
+    var dateChangeLogs = filterLogEntries(
+      [PUMP_LOG_RECORDS.LID_DATE_CHANGED],
+      data.log_records
+    );
+
+    for (var i = 0; i < timeChangeLogs.length; ++i) {
+      var tc = timeChangeLogs[i];
+      var tc_base = sundial.floor(tc.rawTimestamp, 'day').valueOf();
+
+      // look for a corresponding date change event
+      var found = false;
+      var index = tc.index;
+      var lastIndex = data.log_records[data.log_records.length - 1].index;
+      var SEARCH_THRESHOLD = 5;
+      var start = index < SEARCH_THRESHOLD ? 0 : index - SEARCH_THRESHOLD; // look at x entries before
+      var end = (index + SEARCH_THRESHOLD) < lastIndex ?
+        index + SEARCH_THRESHOLD : lastIndex; // and up to x entries after
+      for (var k = start; k <= end; k++) {
+        var event = _.find(dateChangeLogs, {index: k});
```

Comment:
So it occurred to me that this was really all that was needed to have the `_.without` work properly. However, it's not the most efficient approach (but since `dateChangeLogs` is never likely to be a _huge_ array I don't see much of a problem).

With the cluster of 3 events in my data, I ended up just pairing two and having the additional time change just a separate change (a clock drift offset). I think that's actually...fine, but we can discuss. (There did turn out to be additional complications re: having the date & time changes separate.)